client/connect.go

```diff
@@ -158,8 +159,28 @@ func (cli *Client) GetRemoteAccessToken(ctx context.Context, req RemoteAccessTok
 	return &remoteAccessToken, nil
 }
 
+// Connect establishes a connection to a remote devices
+func (cli *Client) Connect(ctx context.Context, connections map[string][]RemoteAccessTarget) error {
```

Comment:
I think we should use a list here (just like you pass in the config file): `[]RemoteAccessConnection`.

If you really want to have fun, you could look into custom [JSON Unmarshaller](https://pkg.go.dev/encoding/json#example-package-CustomMarshalJSON) for the targets, so the:
```go
type RemoteAccessConnection struct {
    // ...
	Targets RemoteAccessTargets `json:"targets"`
}
```

uses a new type:

```go
type RemoteAccessTargets []RemoteAccessTarget
```

which has the custom marshaller implemented:
```go
func (rat *RemoteAccessTargets) UnmarshalJSON(b []byte) error {
	//...
}
```

which looks if you got the targets as a comma-separated-string and if yes, do the parsing there.

But that's only if you want to have some fun before X-mass ;)

Reply:
I'll leave this for now if not strictly necessary